ShopApp/Repositories/UserRepository.cs

```diff
@@ -8,38 +8,46 @@
 
 namespace ShopApp.Repositories
 {
-    internal class UserRepository : IUserRepository
+    public class UserRepository : IUserRepository
     {
         private ShopDbContext _shopDbContext;
         public UserRepository(ShopDbContext context)
         {
             _shopDbContext = context;
         }
-        public async Task<bool> AddUser(User user)
+        public async Task<User> AddUser(User user)
         {
+            if (user is null)
+            {
+                throw new NullReferenceException("User can`t be null");
+            }
             try
             {
                 await _shopDbContext.Users.AddAsync(user);
                 await _shopDbContext.SaveChangesAsync();
-                return true;
+                return _shopDbContext.Users.First(u => u.Id == user.Id);
             }
             catch (DbUpdateException)
             {
-                return false;
+                throw new ArgumentException("Couldn`t add user");
             }
         }
 
-        public async Task<bool> Delete(User user)
+        public async Task<User> Delete(User user)
         {
+            if (user is null)
+            {
+                throw new NullReferenceException("User can`t be null");
+            }
             try
             {
                 _shopDbContext.Users.Remove(user);
                 await _shopDbContext.SaveChangesAsync();
-                return true;
+                return user;
             }
             catch (DbUpdateException)
             {
-                return false;
+                throw new ArgumentException("Couldn`t delete user");
             }
         }
 
```

Comment:
When deleting User, u just using his id. So u get user Id from arguments, checking if it exists in DB using FirstOfDefault. If does not, u throw ArgumentExeption("No such user in DB"). If exists, you removing it from DB